src/extension/infer.rs

```diff
@@ -546,57 +546,51 @@ impl UnificationContext {
     pub fn results(&self) -> Result<ExtensionSolution, InferExtensionError> {
         // Check that all of the metavariables associated with nodes of the
         // graph are solved
+        let dependees = {
+            let mut h: HashMap<Meta, Vec<Meta>> = HashMap::new();
+            for (m, m2) in self.constraints.iter().flat_map(|(m, cs)| {
+                cs.iter().flat_map(|c| match c {
```

Comment:
Hmmm, this one could be a filter_map; or we could do `constraints.iter().flat_map(|(m,cs)| cs.map(|c|(m,c))).filter_map(|(m,c) match c.....)`. "Golf", to some extent